ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -587,16 +595,28 @@ static enum pkcs11_rc create_pub_key_attributes(struct obj_attrs **out,
 
 	switch (get_key_type(*out)) {
 	case PKCS11_CKK_RSA:
-		mandated = rsa_public_key_mandated;
-		opt_or_null = rsa_public_key_opt_or_null;
-		mandated_count = ARRAY_SIZE(rsa_public_key_mandated);
-		opt_or_null_count = ARRAY_SIZE(rsa_public_key_opt_or_null);
+		if (function == PKCS11_FUNCTION_GENERATE_PAIR) {
+			mandated = rsa_pub_key_gen_mand;
+			opt = rsa_pub_key_gen_opt;
+			mandated_count = ARRAY_SIZE(rsa_pub_key_gen_mand);
+			opt_count = ARRAY_SIZE(rsa_pub_key_gen_opt);
+		} else if (function == PKCS11_FUNCTION_IMPORT) {
+			mandated = rsa_pub_key_create_mand;
+			opt = rsa_pub_key_create_opt;
```

Comment:
no need to define an empty table. Keep `opt`/`opt_count` to their default init value.

Reply:
In here I felt that symmetry was more important for clarity than the "optimization". Can remove it thou.

If we go for that then I suppose we should remove `ec_private_key_mandated` too.